src/perlDebug.ts

```diff
@@ -59,11 +61,18 @@ class PerlDebugSession extends LoggingDebugSession {
 
 	private _variableHandles = new Handles<string>();
 
-	private perlDebugger = new perlDebuggerConnection();
+	public dcSupportsRunInTerminal: boolean = false;
+
+	private perlDebugger;
 
 	public constructor() {
 		super('perl_debugger.log');
 
+		// The perlDebuggerConnection needs access to the session to make
+		// reverse requests like `runInTerminal` and to send custom events
+		// to the extension (e.g., to write to some output channel.)
+		this.perlDebugger = new perlDebuggerConnection(this);
```

Comment:
hmm - yeah, I think we should decouple this one passing on `this` will make a circular dependency - making testing etc. more difficult

We should have the `perlDebuggerConnection` have a public method for the session to attach an event listener eg. :
```js
  this.perlDebugger.on('output', (data) => {
    // The session can now do it's thing
  });
```

in the adapter we can setup an eventemitter https://nodejs.org/api/events.html#events_class_eventemitter

We could add events like `debug`, `error` etc. but `output` would be super usefull because I want to add a snapshot of the raw output from perl DB and use snapshot testing for the test matrix.

Reply:
Let me know if you have a different opinion - I'm open for suggestions

Reply:
Hmm. Let's ignore events and messaging for the moment. This is about how to call `runInTerminal`. Basically, `vscode` asks `perlDebug.ts` to launch something, which then asks `adapter.ts` to launch something, which then asks `vscode` to launch something to satisfy `vscode`'s own launch request. So basically we end up telling `vscode` »No, you launch the debug target yourself!«. How can that be done better than in this PR?

Reply:
Now I get it - it's because it needs to setup a perl debug host in the terminal - right?

We should extract the part about setting up the remote session host from `adapter.ts` into `localSessionHost.ts` - it might also be useful when writing tests to be able to setup a host.

(not sure if we should rename`adapter.ts` file to `clientAdapter.ts`)

in the perlDebug.ts file: (could be renamed to `perlDebugExtension.ts` - it's responsible for wiring up vs code and the adapter etc.)
```js
  if (this.dcSupportsRunInTerminal === true) {
    const port = await createLocalSessionHost(options);
    // configure the port for the terminalSession debugger
  }
```